### ISSUE TITLE
After the 60s fallback dismiss fails, the dialog stays pending forever and the watcher lease is never returned

### ISSUE THREAD

**Issue by PM-Shawn** [PM-Shawn/Abu-Cowork#383]
External-review bookkeeping item from batch-3 PR-1 (🟢, not merge-blocking). On one narrow path it silently undoes the "dialog lease" that PR-1 just fixed.

## Symptom

`electron/browserHost.cjs`: the 60-second fallback timer is **one-shot**, and `answerDialog` runs the CDP command first and does its bookkeeping afterwards:

```js
await dbg.sendCommand('Page.handleJavaScriptDialog', …);   // ← throws here
disarmTimer(...);          // never reached
pendingDialogs.delete(id); // never reached
```

When `Page.handleJavaScriptDialog` throws (with the debugger still attached): **the entry stays in `pendingDialogs`, and the timer has already fired and is never re-armed**. From then on `detachDialogWatcherIfIdle` refuses to detach the watcher forever, because `pendingDialogs.has(id)` → dialog interception on that tab becomes **permanent** → the harm PR-1 fixed with the lease comes back: **when the user clicks a confirm box on that page themselves, the native dialog still does not appear**.

## This is not "keep the pending entry" being wrong

"On a failed answer, keep the dialog pending instead of clearing it" is **deliberate and correct** — a pinning test guards it (`a failed answer leaves the dialog pending rather than reporting a freed tab`): reporting an unclosed dialog as "tab freed" would be worse.

The only problem is that **there is no second chance after the failure**. The window is genuinely narrow: the `detach` event clears the entry, and so does `Page.javascriptDialogClosed`, so getting stuck requires "command failed + debugger still attached + dialog not actually closed" all at once.

## Suggested fix (pick one)

- After a failed fallback, **re-arm the timer once**, so the 60-second fallback gets another attempt; or
- **`dbg.detach()` on the second failure**, letting the tab fall back to native dialog behaviour (the user can at least close it themselves).

## Unverified

This came out of reading the code during review; it has **not** been reproduced end to end by forcing the CDP command to fail. Before fixing, write a test where `Page.handleJavaScriptDialog` throws, to confirm or refute it.
